src/go/k8s/controllers/redpanda/cluster_controller_configuration.go

```diff
@@ -81,6 +82,19 @@ func (r *ClusterReconciler) reconcileConfiguration(
 		return errorWithContext(err, "error creating the admin API client")
 	}
 
+	// Checking if the feature is active because in the initial stages of cluster creation, it takes time for the feature to be activated
+	// and the API returns the same error (400) that is returned in case of malformed input, which causes a stop of the reconciliation
+	var centralConfigActive bool
+	if centralConfigActive, err = adminutils.IsFeatureActive(adminAPI, adminutils.CentralConfigFeatureName); err != nil {
+		return errorWithContext(err, "could not determine if central config is active in the cluster")
```

Comment:
Maybe this error could be a bit more helpful and say ", is redpanda >=22.1.1 in use?", as this will be a 404 on older redpandas.

(or if there is some other higher-level check for redpanda version before calling this function, then never mind)

Reply:
Yeah, there's the feature flag check at the beginning that skips the whole reconciliation for older versions: https://github.com/redpanda-data/redpanda/pull/4383/files/8313652c0eeec484b69592cee89ef2bdaa599b39..f0f56aaf8ec5068b1df45b4de5650c5b5caf9255#diff-b7483c0f7a12ac15220119aaf8f47585436c3d7b60a4a6783118f7b3201a3fe4R41